setuptools/_distutils_shim.py

```diff
@@ -74,6 +74,9 @@ def do_override():
 
 
 class DistutilsMetaFinder:
+    # for forensic purposes, since we may not be able to get at the module object easily
+    _fromfile = __file__
```

Comment:
I believe accessing `__file__` may cause problems under a zipimport scenario. That's something we'd want to verify before accepting.